clients/src/main/java/org/apache/kafka/common/metrics/stats/SampledStat.java

```diff
@@ -70,7 +72,7 @@ private Sample advance(MetricConfig config, long timeMs) {
     }
 
     protected Sample newSample(long timeMs) {
-        return new Sample(this.initialValue, timeMs);
+        return this.timeWindowMs != -1 ? new Sample(this.initialValue, timeMs) : new Sample(this.initialValue, timeMs, this.timeWindowMs);
```

Comment:
Sorry, I mistakenly wrote the other way in the commit 
```suggestion
        return this.timeWindowMs > 0 ? new Sample(this.initialValue, timeMs, this.timeWindowMs) : new Sample(this.initialValue, timeMs);
```